tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -212,6 +218,14 @@ class LibraryGenerator {
             Code.scope((a) =>
                 'if(data[\'className\'] == \'${classInfo.className}\'){'
                 'return deserialize<${a(refer(classInfo.className, classInfo.fileRef()))}>(data[\'data\']);}'),
+          if (config.name != 'serverpod' && serverCode)
+            Block.of([
+              Code('if(data[\'className\'].startsWith(\'serverpod.\')){'
+                  'data[\'className\'] = data[\'className\'].substring(${'serverpod'.length + 1});'),
+              Code.scope((a) =>
+                  'return ${a(refer('Protocol', serverpodProtocolUrl(serverCode)))}().deserializeByClassName(data);'),
+              const Code('}'),
```

Comment:
General question on this code. We are mixing a lot of `Code` blocks together with actual builder patterns (for example `refer('variable').property(myMethod).call([])` vs `Code('variable.myMethod()')`). When are you choosing which? My impression so far is that `Code` blocks easily get a bit messy and make code reuse harder. Mixing them I also feel makes it a bit harder to read because of the inconsistent building paradigm (template vs builder).

Reply:
We don't really have a clear strategy here honestly.
Often we just grab what is closest in context and then apply it to our specific scenario.

As far as I understand there are a couple of limitations with the builder pattern that forces us to use code blocks. For example it is not possible to write if statements using the builder pattern, so we are forced to use this middleway.